File: src/main/java/site/hirecruit/hr/domain/worker/repository/WorkerRepository.java
package site.hirecruit.hr.domain.worker.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import site.hirecruit.hr.domain.worker.entity.WorkerEntity;

import java.util.Optional;

public interface WorkerRepository extends JpaRepository<WorkerEntity, Long> {
    boolean existsByGithubId(Long id);

    Optional<WorkerEntity> findByGithubId(long githubId);
}

File: src/main/java/site/hirecruit/hr/domain/worker/repository/WorkerRepository.kt
package site.hirecruit.hr.domain.worker.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import site.hirecruit.hr.domain.worker.dto.WorkerDto
import site.hirecruit.hr.domain.worker.entity.WorkerEntity

@Repository
interface WorkerRepository : JpaRepository<WorkerEntity, Long>, WorkerCustomRepository {

    fun findByUser_GithubId(githubId: Long): WorkerEntity?

    fun findByCompany_CompanyId(companyId: Long): List<WorkerEntity>
}